userspace/libsinsp/user_event.cpp

```diff
@@ -20,6 +20,97 @@ along with sysdig.  If not, see <http://www.gnu.org/licenses/>.
 #include "sinsp_int.h"
 #include "user_event.h"
 
+//
+// event_scope
+//
+
+const std::string event_scope::SCOPE_OP_AND = "and";
+
+// these string lists contain reserved strings; some of the reserved
+// strings are escaped and mandatory to be first in RESERVED_STRINGS
+// and have their escaped counterparts in the REPLACEMENT_STRINGS,
+// in the same order as they appear in RESERVED_STRINGS
+const event_scope::string_list_t event_scope::RESERVED_STRINGS =
+	{"'"/*!must be first!*/, "=", " and ", " "};
+const event_scope::string_list_t event_scope::REPLACEMENT_STRINGS =
+	{"\\'"/*!must be first!*/};
+
+event_scope::event_scope(const std::string& key, const std::string& value)
+{
+	add(key, value, "");
+}
+
+bool event_scope::add(const std::string& key, const std::string& value, const std::string& op)
+{
+	if(check(key))
+	{
+		std::string k(key);
+		std::string o(!m_scope.empty() ? op : "");
+		std::string v(value);
+		replace(v);
+		if(!v.empty())
+		{
+			if(!o.empty())
+			{
+				m_scope.append(1, ' ').append(trim(o)).append(1, ' ');
+			}
+			m_scope.append(trim(k)).append("='").append(trim(v)).append(1, 0x27);
+			return true;
+		}
+	}
+	else
+	{
+		g_logger.log("Scope key has invalid value: [" + key + "], not added to scope.", sinsp_logger::SEV_WARNING);
+	}
+	return false;
+}
+
+string& event_scope::replace(std::string& value)
+{
+	ASSERT(RESERVED_STRINGS.size() >= REPLACEMENT_STRINGS.size());
+
+	if(check(value))
```

Comment:
the value can contain anything inside quotes, we should allow `=`, ` and ` or ` ` signs.